### ISSUE TITLE
Over use of assertion

### ISSUE THREAD

**Issue by Ducasse** [pharo-graphics/Bloc#82]
opacity: aNumber
	self
		assert: [ aNumber isNotNil ]
		description: [ 'Opacity must not be nil' ].
	self
		assert: [ aNumber between: 0.0 and: 1.0 ]
		description: [ 'Opacity value must be within interval [0..1]' ].

	opacity isNotNil
		ifTrue: [ (BlImmutableObjectChangeError object: self) signal ].

	opacity := aNumber

**Comment by Ducasse** [comment]
The use of BlImmutableobjectchange error is not good. 
Better mark the object as immutable.

**Comment by syrel** [comment]
@Ducasse yes, now that Pharo9 has supports (it does, right?) readonly objects it is better to use it rather than errors

**Comment by tinchodias** [comment]
For the immutability check, I opened a new issue (#118). For the others, there was a long discussion.
